### ISSUE TITLE
Handling long text and inline elements

### ISSUE THREAD

**Issue by jarrodldavis** [sveltejs/prettier-plugin-svelte#20]
Currently, the plugin breaks at the start and end of child content when it can't fit on one line:

```diff
--- input.html	2019-05-12 16:49:09.000000000 -0600
+++ output.html	2019-05-12 16:49:09.000000000 -0600
@@ -1 +1,5 @@
-<span>Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur ad fugit cupiditate vel ullam voluptatum maxime harum nemo ipsa sequi sapiente assumenda vitae, beatae iste sed veritatis ipsam dicta deserunt.</span>
+<span>
+    Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur ad fugit
+    cupiditate vel ullam voluptatum maxime harum nemo ipsa sequi sapiente
+    assumenda vitae, beatae iste sed veritatis ipsam dicta deserunt.
+</span>
```

Usually, this isn't a problem, but it can be an issue if the parent element (in the above example, `<span>`) has a `display` CSS property with a value of `inline`. In that case, an extra space will be present between sibling nodes that may be undesirable, especially when surrounded by other inline elements or non-whitespace text.

Another cause of likely unwanted wrapping is when an inline element is present as a child element.

Here is an example, similar to #14:

```diff
--- input.html	2019-05-12 17:17:01.000000000 -0600
+++ output.html	2019-05-12 17:17:01.000000000 -0600
@@ -1,3 +1,5 @@
 <div>
-    [<span>{score}</span>]
+    [
+    <span>{score}</span>
+    ]
 </div>
```

---

Perhaps the easiest way to rectify both issues is to hardcode a list of inline elements when checking if a node is an inline node:

https://github.com/UnwrittenFun/prettier-plugin-svelte/blob/771df545e15f422a9597e9153a1518dbbba8c079/src/print/index.ts#L467-L469

This will cause the element to be treated as part of the inline sequence `fill` groups already used in `printChildren` – this will allow the element's tags to remain in its current position and only break between words when necessary, not forcibly before the start tag and at the start and end of the element's child content.

For example:

```diff
--- input.html	2019-05-12 17:30:45.000000000 -0600
+++ output.html	2019-05-12 17:30:45.000000000 -0600
@@ -1 +1,3 @@
-<span>Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur ad fugit cupiditate vel ullam voluptatum maxime harum nemo ipsa sequi sapiente assumenda vitae, beatae iste sed veritatis ipsam dicta deserunt.</span>
+<span>Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur ad fugit
+    cupiditate vel ullam voluptatum maxime harum nemo ipsa sequi sapiente
+    assumenda vitae, beatae iste sed veritatis ipsam dicta deserunt.</span>
```

This (obviously) has the disadvantage of making assumptions about the the `display` property of all elements, but (in my opinion) should handle most cases where undesired wrapping currently occurs.

---

What is your opinion on this? Is this something you want to try to handle? Should we go with the easy route, or should we try to replicate the (very complicated) logic used for vanilla HTML by Prettier core?

**Comment by UnwrittenFun** [comment]
I think at the very least we need to do the list of inline elements. But having a way to mark an element as inline would probably be useful as well.

**Comment by dummdidumm** [comment]
Fixed through #120
